pysatSeasons/avg.py

```diff
@@ -1,372 +1,459 @@
-                    # get the indicies of those pieces of data in that bin
+                    # Get the indices of those pieces of data in that bin.
                     xindex, = np.where(xind == xi)
+
                     if len(xindex) > 0:
-                        # look up the data along y (label2) at that set of
-                        # indices (a given x)
-                        yData = inst.data.iloc[xindex]
-                        # digitize that, to sort data into bins along y
+                        # Look up the data along y (label2) at that set of
+                        # indices (a given x).
+                        yinst.data = inst[xindex]
+
+                        # Digitize that, to sort data into bins along y
                         # (label2) (get bin indexes)
-                        yind = np.digitize(yData[label2], biny) - 1
-                        # for each possible y index
+                        yind = np.digitize(yinst[label2], biny) - 1
+
+                        # For each possible y index
                         for yj in yarr:
-                            # select data with this y index (and we already
+                            # Select data with this y index (and we already
                             # filtered for this x index)
                             yindex, = np.where(yind == yj)
+
                             if len(yindex) > 0:
-                                # for each data product label zk
+                                # For each data product label zk
                                 for zk in zarr:
-                                    # take the data (already filtered by x)
-                                    # filter it by y and
-                                    # select the data product, put it in a
-                                    # list, and extend the deque
-                                    indlab = yData.columns.get_loc(data_label[zk])
-                                    ans[zk][yj][xi].extend(yData.iloc[yindex,
-                                                                      indlab].tolist())
+                                    # Take the data (already filtered by x),
+                                    # filter it by y, select the data product,
+                                    # put it in a list, and extend the deque.
+                                    ans[zk][yj][xi].extend(
+                                        yinst[yindex, data_label[zk]])
 
     return _calc_2d_median(ans, data_label, binx, biny, xarr, yarr, zarr,
-                           numx, numy, numz, returnData)
+                           numx, numy, numz, returnData, return_data)
 
 
 def _calc_2d_median(ans, data_label, binx, biny, xarr, yarr, zarr, numx,
-                    numy, numz, returnData=False):
+                    numy, numz, returnData=None, return_data=False):
+    """Return a 2D average of nD `data_label` over season and `label1` `label2`.
```

Comment:
```suggestion
    """Calculate a 2D median of nD `data_label` over time.
```

Unless label1 and label2 are involved somehow?  The statement doesn't make much sense so I don't know if my correction is any good.

Reply:
Updated to include binx and biny, the binning variable info